File: performance-comparison/benchmarks/numerical/clojure/primes.clj
(defn count-primes [limit]
  (let [sieve (boolean-array (inc limit))]
    (loop [i 2 count 0]
      (cond
        (> i limit) count
        (aget sieve i) (recur (inc i) count)
        :else (do
                (loop [j (* 2 i)]
                  (when (<= j limit)
                    (aset sieve j true)
                    (recur (+ j i))))
                (recur (inc i) (inc count)))))))

(let [n (Long/parseLong (first *command-line-args*))]
  (println (count-primes n)))

File: performance-comparison/benchmarks/numerical/python/primes.py
import sys

def count_primes(limit):
    sieve = bytearray(limit + 1)
    count = 0
    for i in range(2, limit + 1):
        if not sieve[i]:
            count += 1
            for j in range(2 * i, limit + 1, i):
                sieve[j] = 1
    return count

n = int(sys.argv[1]) if len(sys.argv) > 1 else 10000
print(count_primes(n))
